Sources/ArgumentParser/Utilities/Platform.swift

```diff
@@ -120,16 +120,57 @@ extension Platform {
     (80, 25)
   }
   
-  /// Returns the current terminal size, or the default if the size is
-  /// unavailable.
-  static func terminalSize() -> (width: Int, height: Int) {
+  /// The terminal size specified by the COLUMNS and LINES overrides
+  /// (if present).
+  ///
+  /// Per the [Linux environ(7) manpage][linenv]:
+  ///
+  /// ```
+  /// * COLUMNS and LINES tell applications about the window size,
+  ///   possibly overriding the actual size.
+  /// ```
+  ///
+  /// And the [FreeBSD environ(7) version][bsdenv]:
+  ///
+  /// ```
+  /// COLUMNS    The user's preferred width in column positions for the
+  ///            terminal.  Utilities such as ls(1) and who(1) use this
+  ///            to format output into columns.   If  unset  or  empty,
+  ///            utilities  will use an ioctl(2) call to ask the termi-
+  ///            nal driver for the width.
+  /// ```
+  ///
+  /// > Note: Always ignored on Windows.
+  ///
+  /// [linenv]: https://man7.org/linux/man-pages/man7/environ.7.html:~:text=COLUMNS
+  /// [bsdenv]: https://man.freebsd.org/cgi/man.cgi?environ(7)#:~:text=COLUMNS
+  static func overridenTerminalSize() -> (width: Int, height: Int) {
```

Comment:
Could we use `Int?`/`nil` instead of `0`?

Reply:
I originally went with `0` based on the implementation of the `TIOCGWINSZ` logic, but now that you point it out, in retrospect I realize that the use of `0` was based on the semantics of the `ioctl()` itself rather than informing the code as a whole 😅. Will push an update in a few minutes!